src/libs/nav.js

```diff
@@ -43,15 +43,10 @@ export const goToPath = (...args) => {
   history.push({ pathname: getPath(...args) })
 }
 
-export class Redirector extends Component {
-  componentDidMount() {
-    const { pathname, search } = this.props
-    history.replace({ pathname, search })
-  }
+export const Redirector = ({ pathname, search }) => {
+  useOnMount(() => history.replace({ pathname, search }))
```

Comment:
Writing this as a single-expression arrow function means that the result of `replace` is returned. Since this happens to be `undefined`, it works as expected, but if it returned something else React would try to register it as a callback on unmount. Since that's not what we intend here, I think it would be clearer to write it like this to be clear we're not returning anything:
```suggestion
  useOnMount(() => {
    history.replace({ pathname, search })
  })
```